2021-03-10 08:49:04.302 [main] INFO  Application - Autoreload is disabled because the development mode is off.
2021-03-10 08:49:04.810 [main] INFO  [Koin] - loaded 4 definitions - 2.894782 ms
2021-03-10 08:49:04.828 [main] DEBUG com.zaxxer.hikari.HikariConfig - Driver class com.mysql.cj.jdbc.Driver found in Thread context class loader jdk.internal.loader.ClassLoaders$AppClassLoader@2c13da15
2021-03-10 08:49:04.840 [main] DEBUG com.zaxxer.hikari.HikariConfig - HikariPool-1 - configuration:
2021-03-10 08:49:04.856 [main] DEBUG com.zaxxer.hikari.HikariConfig - allowPoolSuspension.............false
2021-03-10 08:49:04.856 [main] DEBUG com.zaxxer.hikari.HikariConfig - autoCommit......................false
2021-03-10 08:49:04.856 [main] DEBUG com.zaxxer.hikari.HikariConfig - catalog.........................none
2021-03-10 08:49:04.856 [main] DEBUG com.zaxxer.hikari.HikariConfig - connectionInitSql...............none
2021-03-10 08:49:04.856 [main] DEBUG com.zaxxer.hikari.HikariConfig - connectionTestQuery.............none
2021-03-10 08:49:04.857 [main] DEBUG com.zaxxer.hikari.HikariConfig - connectionTimeout...............30000
2021-03-10 08:49:04.857 [main] DEBUG com.zaxxer.hikari.HikariConfig - dataSource......................none
2021-03-10 08:49:04.857 [main] DEBUG com.zaxxer.hikari.HikariConfig - dataSourceClassName.............none
2021-03-10 08:49:04.857 [main] DEBUG com.zaxxer.hikari.HikariConfig - dataSourceJNDI..................none
2021-03-10 08:49:04.858 [main] DEBUG com.zaxxer.hikari.HikariConfig - dataSourceProperties............{maintainTimeStats=false, password=<masked>, prepStmtCacheSqlLimit=2048, rewriteBatchedStatements=true, elideSetAutoCommits=true, cachePrepStmts=true, cacheResultSetMetadata=true, prepStmtCacheSize=250, useServerPrepStmts=true, useLocalSessionState=true, cacheServerConfiguration=true}
2021-03-10 08:49:04.858 [main] DEBUG com.zaxxer.hikari.HikariConfig - driverClassName................."com.mysql.cj.jdbc.Driver"
2021-03-10 08:49:04.858 [main] DEBUG com.zaxxer.hikari.HikariConfig - healthCheckProperties...........{}
2021-03-10 08:49:04.858 [main] DEBUG com.zaxxer.hikari.HikariConfig - healthCheckRegistry.............none
2021-03-10 08:49:04.858 [main] DEBUG com.zaxxer.hikari.HikariConfig - idleTimeout.....................600000
2021-03-10 08:49:04.859 [main] DEBUG com.zaxxer.hikari.HikariConfig - initializationFailTimeout.......1
2021-03-10 08:49:04.859 [main] DEBUG com.zaxxer.hikari.HikariConfig - isolateInternalQueries..........false
2021-03-10 08:49:04.860 [main] DEBUG com.zaxxer.hikari.HikariConfig - jdbcUrl.........................jdbc:mysql://localhost:3308/chucknorris?useUnicode=true&characterEncoding=UTF-8
2021-03-10 08:49:04.860 [main] DEBUG com.zaxxer.hikari.HikariConfig - leakDetectionThreshold..........0
2021-03-10 08:49:04.860 [main] DEBUG com.zaxxer.hikari.HikariConfig - maxLifetime.....................1800000
2021-03-10 08:49:04.860 [main] DEBUG com.zaxxer.hikari.HikariConfig - maximumPoolSize.................3
2021-03-10 08:49:04.860 [main] DEBUG com.zaxxer.hikari.HikariConfig - metricRegistry..................none
2021-03-10 08:49:04.860 [main] DEBUG com.zaxxer.hikari.HikariConfig - metricsTrackerFactory...........none
2021-03-10 08:49:04.860 [main] DEBUG com.zaxxer.hikari.HikariConfig - minimumIdle.....................3
2021-03-10 08:49:04.860 [main] DEBUG com.zaxxer.hikari.HikariConfig - password........................<masked>
2021-03-10 08:49:04.860 [main] DEBUG com.zaxxer.hikari.HikariConfig - poolName........................"HikariPool-1"
2021-03-10 08:49:04.861 [main] DEBUG com.zaxxer.hikari.HikariConfig - readOnly........................false
2021-03-10 08:49:04.861 [main] DEBUG com.zaxxer.hikari.HikariConfig - registerMbeans..................false
2021-03-10 08:49:04.861 [main] DEBUG com.zaxxer.hikari.HikariConfig - scheduledExecutor...............none
2021-03-10 08:49:04.861 [main] DEBUG com.zaxxer.hikari.HikariConfig - schema..........................none
2021-03-10 08:49:04.861 [main] DEBUG com.zaxxer.hikari.HikariConfig - threadFactory...................internal
2021-03-10 08:49:04.861 [main] DEBUG com.zaxxer.hikari.HikariConfig - transactionIsolation............"TRANSACTION_REPEATABLE_READ"
2021-03-10 08:49:04.861 [main] DEBUG com.zaxxer.hikari.HikariConfig - username........................"root"
2021-03-10 08:49:04.861 [main] DEBUG com.zaxxer.hikari.HikariConfig - validationTimeout...............5000
2021-03-10 08:49:04.863 [main] DEBUG com.zaxxer.hikari.HikariConfig - HikariPool-1 - configuration:
2021-03-10 08:49:04.865 [main] DEBUG com.zaxxer.hikari.HikariConfig - allowPoolSuspension.............false
2021-03-10 08:49:04.865 [main] DEBUG com.zaxxer.hikari.HikariConfig - autoCommit......................false
2021-03-10 08:49:04.865 [main] DEBUG com.zaxxer.hikari.HikariConfig - catalog.........................none
2021-03-10 08:49:04.865 [main] DEBUG com.zaxxer.hikari.HikariConfig - connectionInitSql...............none
2021-03-10 08:49:04.865 [main] DEBUG com.zaxxer.hikari.HikariConfig - connectionTestQuery.............none
2021-03-10 08:49:04.865 [main] DEBUG com.zaxxer.hikari.HikariConfig - connectionTimeout...............30000
2021-03-10 08:49:04.865 [main] DEBUG com.zaxxer.hikari.HikariConfig - dataSource......................none
2021-03-10 08:49:04.866 [main] DEBUG com.zaxxer.hikari.HikariConfig - dataSourceClassName.............none
2021-03-10 08:49:04.866 [main] DEBUG com.zaxxer.hikari.HikariConfig - dataSourceJNDI..................none
2021-03-10 08:49:04.866 [main] DEBUG com.zaxxer.hikari.HikariConfig - dataSourceProperties............{maintainTimeStats=false, password=<masked>, prepStmtCacheSqlLimit=2048, rewriteBatchedStatements=true, elideSetAutoCommits=true, cachePrepStmts=true, cacheResultSetMetadata=true, prepStmtCacheSize=250, useServerPrepStmts=true, useLocalSessionState=true, cacheServerConfiguration=true}
2021-03-10 08:49:04.866 [main] DEBUG com.zaxxer.hikari.HikariConfig - driverClassName................."com.mysql.cj.jdbc.Driver"
2021-03-10 08:49:04.866 [main] DEBUG com.zaxxer.hikari.HikariConfig - healthCheckProperties...........{}
2021-03-10 08:49:04.866 [main] DEBUG com.zaxxer.hikari.HikariConfig - healthCheckRegistry.............none
2021-03-10 08:49:04.866 [main] DEBUG com.zaxxer.hikari.HikariConfig - idleTimeout.....................600000
2021-03-10 08:49:04.866 [main] DEBUG com.zaxxer.hikari.HikariConfig - initializationFailTimeout.......1
2021-03-10 08:49:04.866 [main] DEBUG com.zaxxer.hikari.HikariConfig - isolateInternalQueries..........false
2021-03-10 08:49:04.867 [main] DEBUG com.zaxxer.hikari.HikariConfig - jdbcUrl.........................jdbc:mysql://localhost:3308/chucknorris?useUnicode=true&characterEncoding=UTF-8
2021-03-10 08:49:04.867 [main] DEBUG com.zaxxer.hikari.HikariConfig - leakDetectionThreshold..........0
2021-03-10 08:49:04.867 [main] DEBUG com.zaxxer.hikari.HikariConfig - maxLifetime.....................1800000
2021-03-10 08:49:04.867 [main] DEBUG com.zaxxer.hikari.HikariConfig - maximumPoolSize.................3
2021-03-10 08:49:04.867 [main] DEBUG com.zaxxer.hikari.HikariConfig - metricRegistry..................none
2021-03-10 08:49:04.867 [main] DEBUG com.zaxxer.hikari.HikariConfig - metricsTrackerFactory...........none
2021-03-10 08:49:04.867 [main] DEBUG com.zaxxer.hikari.HikariConfig - minimumIdle.....................3
2021-03-10 08:49:04.867 [main] DEBUG com.zaxxer.hikari.HikariConfig - password........................<masked>
2021-03-10 08:49:04.868 [main] DEBUG com.zaxxer.hikari.HikariConfig - poolName........................"HikariPool-1"
2021-03-10 08:49:04.868 [main] DEBUG com.zaxxer.hikari.HikariConfig - readOnly........................false
2021-03-10 08:49:04.868 [main] DEBUG com.zaxxer.hikari.HikariConfig - registerMbeans..................false
2021-03-10 08:49:04.868 [main] DEBUG com.zaxxer.hikari.HikariConfig - scheduledExecutor...............none
2021-03-10 08:49:04.868 [main] DEBUG com.zaxxer.hikari.HikariConfig - schema..........................none
2021-03-10 08:49:04.868 [main] DEBUG com.zaxxer.hikari.HikariConfig - threadFactory...................internal
2021-03-10 08:49:04.868 [main] DEBUG com.zaxxer.hikari.HikariConfig - transactionIsolation............"TRANSACTION_REPEATABLE_READ"
2021-03-10 08:49:04.868 [main] DEBUG com.zaxxer.hikari.HikariConfig - username........................"root"
2021-03-10 08:49:04.868 [main] DEBUG com.zaxxer.hikari.HikariConfig - validationTimeout...............5000
2021-03-10 08:49:04.871 [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Starting...
2021-03-10 08:49:05.012 [main] DEBUG com.zaxxer.hikari.pool.PoolBase - HikariPool-1 - Failed to create/setup connection: Communications link failure
2021-03-10 08:49:05.016 [main] DEBUG com.zaxxer.hikari.pool.HikariPool - HikariPool-1 - Cannot acquire connection from data source
com.mysql.cj.jdbc.exceptions.CommunicationsException: Communications link failure
	at com.mysql.cj.jdbc.exceptions.SQLError.createCommunicationsException(SQLError.java:174)
	at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:64)
	at com.mysql.cj.jdbc.ConnectionImpl.createNewIO(ConnectionImpl.java:836)
	at com.mysql.cj.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:456)
	at com.mysql.cj.jdbc.ConnectionImpl.getInstance(ConnectionImpl.java:246)
	at com.mysql.cj.jdbc.NonRegisteringDriver.connect(NonRegisteringDriver.java:197)
	at com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:138)
	at com.zaxxer.hikari.pool.PoolBase.newConnection(PoolBase.java:353)
	at com.zaxxer.hikari.pool.PoolBase.newPoolEntry(PoolBase.java:201)
	at com.zaxxer.hikari.pool.HikariPool.createPoolEntry(HikariPool.java:473)
	at com.zaxxer.hikari.pool.HikariPool.checkFailFast(HikariPool.java:562)
	at com.zaxxer.hikari.pool.HikariPool.<init>(HikariPool.java:115)
	at com.zaxxer.hikari.HikariDataSource.<init>(HikariDataSource.java:81)
	at com.prof18.ktor.chucknorris.sample.database.DatabaseFactoryImpl.hikari(DatabaseFactoryImpl.kt:44)
	at com.prof18.ktor.chucknorris.sample.database.DatabaseFactoryImpl.connect(DatabaseFactoryImpl.kt:18)
	at com.prof18.ktor.chucknorris.sample.ApplicationKt.module(Application.kt:51)
	at com.prof18.ktor.chucknorris.sample.ApplicationKt.module$default(Application.kt:34)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at kotlin.reflect.jvm.internal.calls.CallerImpl$Method.callMethod(CallerImpl.kt:97)
	at kotlin.reflect.jvm.internal.calls.CallerImpl$Method$Static.call(CallerImpl.kt:106)
	at kotlin.reflect.jvm.internal.KCallableImpl.callDefaultMethod$kotlin_reflection(KCallableImpl.kt:173)
	at kotlin.reflect.jvm.internal.KCallableImpl.callBy(KCallableImpl.kt:112)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading.callFunctionWithInjection(ApplicationEngineEnvironmentReloading.kt:406)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading.executeModuleFunction(ApplicationEngineEnvironmentReloading.kt:360)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading.access$executeModuleFunction(ApplicationEngineEnvironmentReloading.kt:35)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading$launchModuleByName$1.invoke(ApplicationEngineEnvironmentReloading.kt:311)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading$launchModuleByName$1.invoke(ApplicationEngineEnvironmentReloading.kt:35)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading.avoidingDoubleStartupFor(ApplicationEngineEnvironmentReloading.kt:335)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading.launchModuleByName(ApplicationEngineEnvironmentReloading.kt:310)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading.access$launchModuleByName(ApplicationEngineEnvironmentReloading.kt:35)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading$instantiateAndConfigureApplication$1.invoke(ApplicationEngineEnvironmentReloading.kt:299)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading$instantiateAndConfigureApplication$1.invoke(ApplicationEngineEnvironmentReloading.kt:35)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading.avoidingDoubleStartup(ApplicationEngineEnvironmentReloading.kt:317)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading.instantiateAndConfigureApplication(ApplicationEngineEnvironmentReloading.kt:297)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading.createApplication(ApplicationEngineEnvironmentReloading.kt:138)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading.start(ApplicationEngineEnvironmentReloading.kt:270)
	at io.ktor.server.netty.NettyApplicationEngine.start(NettyApplicationEngine.kt:160)
	at io.ktor.server.netty.EngineMain.main(EngineMain.kt:26)
	at com.prof18.ktor.chucknorris.sample.ApplicationKt.main(Application.kt:25)
Caused by: com.mysql.cj.exceptions.CJCommunicationsException: Communications link failure
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at com.mysql.cj.exceptions.ExceptionFactory.createException(ExceptionFactory.java:61)
	at com.mysql.cj.exceptions.ExceptionFactory.createException(ExceptionFactory.java:105)
	at com.mysql.cj.exceptions.ExceptionFactory.createException(ExceptionFactory.java:151)
	at com.mysql.cj.exceptions.ExceptionFactory.createCommunicationsException(ExceptionFactory.java:167)
	at com.mysql.cj.protocol.a.NativeSocketConnection.connect(NativeSocketConnection.java:91)
	at com.mysql.cj.NativeSession.connect(NativeSession.java:144)
	at com.mysql.cj.jdbc.ConnectionImpl.connectOneTryOnly(ConnectionImpl.java:956)
	at com.mysql.cj.jdbc.ConnectionImpl.createNewIO(ConnectionImpl.java:826)
	... 39 common frames omitted
Caused by: java.net.ConnectException: Connection refused (Connection refused)
	at java.base/java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.base/java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:399)
	at java.base/java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:242)
	at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:224)
	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:403)
	at java.base/java.net.Socket.connect(Socket.java:609)
	at com.mysql.cj.protocol.StandardSocketFactory.connect(StandardSocketFactory.java:155)
	at com.mysql.cj.protocol.a.NativeSocketConnection.connect(NativeSocketConnection.java:65)
	... 42 common frames omitted
2021-03-10 08:49:05.017 [main] ERROR com.zaxxer.hikari.pool.HikariPool - HikariPool-1 - Exception during pool initialization.
com.mysql.cj.jdbc.exceptions.CommunicationsException: Communications link failure
	at com.mysql.cj.jdbc.exceptions.SQLError.createCommunicationsException(SQLError.java:174)
	at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:64)
	at com.mysql.cj.jdbc.ConnectionImpl.createNewIO(ConnectionImpl.java:836)
	at com.mysql.cj.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:456)
	at com.mysql.cj.jdbc.ConnectionImpl.getInstance(ConnectionImpl.java:246)
	at com.mysql.cj.jdbc.NonRegisteringDriver.connect(NonRegisteringDriver.java:197)
	at com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:138)
	at com.zaxxer.hikari.pool.PoolBase.newConnection(PoolBase.java:353)
	at com.zaxxer.hikari.pool.PoolBase.newPoolEntry(PoolBase.java:201)
	at com.zaxxer.hikari.pool.HikariPool.createPoolEntry(HikariPool.java:473)
	at com.zaxxer.hikari.pool.HikariPool.checkFailFast(HikariPool.java:562)
	at com.zaxxer.hikari.pool.HikariPool.<init>(HikariPool.java:115)
	at com.zaxxer.hikari.HikariDataSource.<init>(HikariDataSource.java:81)
	at com.prof18.ktor.chucknorris.sample.database.DatabaseFactoryImpl.hikari(DatabaseFactoryImpl.kt:44)
	at com.prof18.ktor.chucknorris.sample.database.DatabaseFactoryImpl.connect(DatabaseFactoryImpl.kt:18)
	at com.prof18.ktor.chucknorris.sample.ApplicationKt.module(Application.kt:51)
	at com.prof18.ktor.chucknorris.sample.ApplicationKt.module$default(Application.kt:34)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at kotlin.reflect.jvm.internal.calls.CallerImpl$Method.callMethod(CallerImpl.kt:97)
	at kotlin.reflect.jvm.internal.calls.CallerImpl$Method$Static.call(CallerImpl.kt:106)
	at kotlin.reflect.jvm.internal.KCallableImpl.callDefaultMethod$kotlin_reflection(KCallableImpl.kt:173)
	at kotlin.reflect.jvm.internal.KCallableImpl.callBy(KCallableImpl.kt:112)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading.callFunctionWithInjection(ApplicationEngineEnvironmentReloading.kt:406)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading.executeModuleFunction(ApplicationEngineEnvironmentReloading.kt:360)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading.access$executeModuleFunction(ApplicationEngineEnvironmentReloading.kt:35)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading$launchModuleByName$1.invoke(ApplicationEngineEnvironmentReloading.kt:311)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading$launchModuleByName$1.invoke(ApplicationEngineEnvironmentReloading.kt:35)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading.avoidingDoubleStartupFor(ApplicationEngineEnvironmentReloading.kt:335)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading.launchModuleByName(ApplicationEngineEnvironmentReloading.kt:310)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading.access$launchModuleByName(ApplicationEngineEnvironmentReloading.kt:35)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading$instantiateAndConfigureApplication$1.invoke(ApplicationEngineEnvironmentReloading.kt:299)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading$instantiateAndConfigureApplication$1.invoke(ApplicationEngineEnvironmentReloading.kt:35)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading.avoidingDoubleStartup(ApplicationEngineEnvironmentReloading.kt:317)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading.instantiateAndConfigureApplication(ApplicationEngineEnvironmentReloading.kt:297)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading.createApplication(ApplicationEngineEnvironmentReloading.kt:138)
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading.start(ApplicationEngineEnvironmentReloading.kt:270)
	at io.ktor.server.netty.NettyApplicationEngine.start(NettyApplicationEngine.kt:160)
	at io.ktor.server.netty.EngineMain.main(EngineMain.kt:26)
	at com.prof18.ktor.chucknorris.sample.ApplicationKt.main(Application.kt:25)
Caused by: com.mysql.cj.exceptions.CJCommunicationsException: Communications link failure
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at com.mysql.cj.exceptions.ExceptionFactory.createException(ExceptionFactory.java:61)
	at com.mysql.cj.exceptions.ExceptionFactory.createException(ExceptionFactory.java:105)
	at com.mysql.cj.exceptions.ExceptionFactory.createException(ExceptionFactory.java:151)
	at com.mysql.cj.exceptions.ExceptionFactory.createCommunicationsException(ExceptionFactory.java:167)
	at com.mysql.cj.protocol.a.NativeSocketConnection.connect(NativeSocketConnection.java:91)
	at com.mysql.cj.NativeSession.connect(NativeSession.java:144)
	at com.mysql.cj.jdbc.ConnectionImpl.connectOneTryOnly(ConnectionImpl.java:956)
	at com.mysql.cj.jdbc.ConnectionImpl.createNewIO(ConnectionImpl.java:826)
	... 39 common frames omitted
Caused by: java.net.ConnectException: Connection refused (Connection refused)
	at java.base/java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.base/java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:399)
	at java.base/java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:242)
	at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:224)
	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:403)
	at java.base/java.net.Socket.connect(Socket.java:609)
	at com.mysql.cj.protocol.StandardSocketFactory.connect(StandardSocketFactory.java:155)
	at com.mysql.cj.protocol.a.NativeSocketConnection.connect(NativeSocketConnection.java:65)
	... 42 common frames omitted
